internal/storage/v1/elasticsearch/spanstore/reader_test.go

```diff
@@ -1247,52 +1198,13 @@ func TestSpanReader_GetEmptyIndex(t *testing.T) {
 	})
 }
 
-func TestSpanReader_ArchiveTraces(t *testing.T) {
-	testCases := []struct {
-		useAliases bool
-		suffix     string
-		expected   string
-	}{
-		{false, "", "jaeger-span-"},
-		{true, "", "jaeger-span-read"},
-		{false, "foobar", "jaeger-span-"},
-		{true, "foobar", "jaeger-span-foobar"},
-	}
-
-	for _, tc := range testCases {
-		t.Run(fmt.Sprintf("useAliases=%v suffix=%s", tc.useAliases, tc.suffix), func(t *testing.T) {
-			withArchiveSpanReader(t, tc.useAliases, tc.suffix, func(r *spanReaderTest) {
-				mockSearchService(r).
-					Return(&elastic.SearchResult{}, nil)
-				mockArchiveMultiSearchService(r, tc.expected).
-					Return(&elastic.MultiSearchResult{
-						Responses: []*elastic.SearchResult{},
-					}, nil)
-				query := spanstore.GetTraceParameters{}
-				trace, err := r.reader.GetTrace(context.Background(), query)
-				require.NotEmpty(t, r.traceBuffer.GetSpans(), "Spans recorded")
-				require.Nil(t, trace)
-				require.EqualError(t, err, "trace not found")
-			})
-		})
-	}
-}
-
-func TestConvertTraceIDsStringsToModels(t *testing.T) {
-	ids, err := toModelTraceIDs([]dbmodel.TraceID{"1", "2", "01", "02", "001", "002"})
-	require.NoError(t, err)
-	assert.Equal(t, []model.TraceID{model.NewTraceID(0, 1), model.NewTraceID(0, 2)}, ids)
-	_, err = toModelTraceIDs([]dbmodel.TraceID{"1", "2", "01", "02", "001", "002", "blah"})
-	require.Error(t, err)
-}
-
 func TestBuildTraceByIDQuery(t *testing.T) {
 	uintMax := ^uint64(0)
-	traceIDNoHigh := model.NewTraceID(0, 1)
-	traceIDHigh := model.NewTraceID(1, 1)
-	traceID := model.NewTraceID(uintMax, uintMax)
+	traceIDNoHigh := newTraceID(0, 1)
+	traceIDHigh := newTraceID(1, 1)
+	traceId := newTraceID(uintMax, uintMax)
```

Comment:
why can't it just be a string?
```suggestion
	traceId := "test-trace-id"
```

Reply:
We can do that for the case when legacy ids are turned off, but for the case when they are enabled we need to ensure that a proper trace id is passed, as high and low of trace id are extracted in this function and used in the query!